clients/roscpp/src/libros/poll_set.cpp

```diff
@@ -177,28 +186,28 @@ void PollSet::update(int poll_timeout)
   createNativePollset();
 
   // Poll across the sockets we're servicing
-  int ret;
-  size_t ufds_count = ufds_.size();
-  if((ret = poll_sockets(&ufds_.front(), ufds_count, poll_timeout)) < 0)
+  boost::shared_ptr<std::vector<socket_pollfd> > ofds = poll_sockets(epfd_, &ufds_.front(), ufds_.size(), poll_timeout);
+  if (!ofds)
   {
-	  ROS_ERROR_STREAM("poll failed with error " << last_socket_error_string());
-    }
-  else if (ret > 0)  // ret = 0 implies the poll timed out, nothing to do
+    ROS_ERROR_STREAM("poll failed with error " << last_socket_error_string());
+  }
+  else
   {
-    // We have one or more sockets to service
-    for(size_t i=0; i<ufds_count; i++)
+    for (std::vector<socket_pollfd>::iterator it = ofds->begin() ; it != ofds->end(); ++it)
     {
-      if (ufds_[i].revents == 0)
-      {
-        continue;
-      }
-
+      int fd = it->fd;
+      int revents = it->revents;
       SocketUpdateFunc func;
       TransportPtr transport;
       int events = 0;
+
+      if (revents == 0)
+      {
+        continue;
+      }
```

Comment:
Should this check / continue stay above the declaration of `func`, `transport` ad `events` as it was before?

Reply:
It can but it really has very little impact since we are talking about variable declaration. I do prefer to have variable declaration at the top of the scope block for clarity purposes. But that's just me.

Reply:
It can stay as-is in this case. I usually aim to keep patches minimal. While there is for sure plenty of code which would benefit cleanup any change makes the review more difficult, has a chance of regressions, and makes backporting patches more effort.

Reply:
makes sense. Thanks.